tree.go

```diff
@@ -249,7 +262,7 @@ func (n *InternalNode) Children() []VerkleNode {
 }
 
 func (n *InternalNode) SetChild(i int, c VerkleNode) error {
-	if i >= NodeWidth-1 {
+	if i >= NodeWidth {
```

Comment:
I forgot to mention: I found this bug the hard way while doing the helpers and running stuff converting trees...

I think we got lucky to not see this bug happening before.